tests/bluetooth/host/keys/bt_keys_add_type/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+if (NOT BOARD STREQUAL unit_testing)
+  message(FATAL_ERROR "This project can only be used with '-DBOARD=unit_testing'.")
+endif()
+
+FILE(GLOB SOURCES src/*.c)
```

Comment:
Looking at all the cmake projects under keys, they are all very similar and each seems to be implementing only a single test, (add/clear/find) but creating a separate main.c and hence a separate test executable. It's a lot of build overhead and execution overhead and I don't think it's adding much.

It seems that if the keys modules is in use, all of these tests are relevant, in which case there only needs to be one CMakeList.txt, prj.conf, and main.c for all of the tests related to keys. There's a lot of duplication here which could be reduced.

Reply:
Your concern is totally valid and I agree with you.
I think we can discuss that point with Nordic team when we meet.

Actually, I think that I combined different tests together at the beginning, then they suggested creating a dedicated UT project for each API.